dandi/download.py

```diff
@@ -706,16 +707,41 @@ def _download_file(
 
     # TODO: how do we discover the total size????
     # TODO: do not do it in-place, but rather into some "hidden" file
+    downloaded = 0
+    prev_digest = None
     for attempt in range(3):
         try:
-            downloaded = 0
-            if digester:
-                downloaded_digest = digester()  # start empty
+            if downloaded:
+                downloaded = max(downloaded - 2 * MAX_CHUNK_SIZE, 0)
             warned = False
             # I wonder if we could make writing async with downloader
-            with open(path, "wb") as writer:
-                for block in downloader():
+            with open(path, "a+b") as writer:
+                first = True
+                for block in downloader(start_at=downloaded):
+                    if first:
+                        writer.seek(downloaded)
+                        if downloaded:
+                            blob1 = writer.read(len(block))
+                            if blob1 == block:
+                                downloaded += len(block)
+                                lgr.debug("Resuming earlier download")
+                                if prev_digest:
+                                    # Set digest state one block back from last
+                                    # run
+                                    downloaded_digest = prev_digest
+                                else:
+                                    assert digester is None
+                                continue
+                            else:
+                                downloaded = 0
+                                downloaded_digest = digester()
+                                lgr.debug("Could not resume earlier download")
```

Comment:
```suggestion
                                lgr.debug("Could not resume earlier download. Will possibly retry from the beginning")
```
so we are not left wondering... could be improved with factual "will" or not restart.  Then please place `3` into a variable (`nattempts`?) so there is no hardcoded values in the code if you decide to compare if that is the last one...